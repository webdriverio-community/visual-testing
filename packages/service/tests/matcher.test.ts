import { describe, it, vi, expect } from 'vitest'
import {
    toMatchScreenSnapshot, toMatchFullPageSnapshot,
    toMatchElementSnapshot, toMatchTabbablePageSnapshot
} from '../src/matcher.js'

const browser = {
    checkScreen: vi.fn().mockResolvedValue(123),
    checkFullPageScreen: vi.fn().mockResolvedValue(123),
    checkElement: vi.fn().mockResolvedValue(123),
    checkTabbablePage: vi.fn().mockResolvedValue(123)
} as any as WebdriverIO.Browser

describe('custom visual matcher', () => {
    it('toMatchScreenSnapshot', async () => {
        await expect(toMatchScreenSnapshot(browser, 'foo', 123, {})).resolves.toEqual({
            pass: true,
            message: expect.any(Function)
        })
        expect(browser.checkScreen).toBeCalledTimes(1)
    })

    it('toMatchFullPageSnapshot', async () => {
        await expect(toMatchFullPageSnapshot(browser, 'foo', 123, {})).resolves.toEqual({
            pass: true,
            message: expect.any(Function)
        })
        expect(browser.checkFullPageScreen).toBeCalledTimes(1)
    })

    it('toMatchElementSnapshot', async () => {
        await expect(toMatchElementSnapshot(browser as any as WebdriverIO.Element, 'foo', 123, {})).resolves.toEqual({
            pass: true,
            message: expect.any(Function)
        })
        expect(browser.checkElement).toBeCalledTimes(1)
    })

    it('toMatchTabbablePageSnapshot', async () => {
        await expect(toMatchTabbablePageSnapshot(browser, 'foo', 123, {})).resolves.toEqual({
            pass: true,
            message: expect.any(Function)
        })
        expect(browser.checkTabbablePage).toBeCalledTimes(1)
    })
})
