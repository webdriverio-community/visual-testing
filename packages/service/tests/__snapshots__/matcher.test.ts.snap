// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`custom visual matcher > defaults to 0 if no expected result is given 1`] = `
"Expected image to have a mismatch percentage of 0%, but was 123%
Please compare the images manually and update the baseline if the new screenshot is correct.

Baseline: foo
Actual Screenshot: bar
Difference: baz

See https://webdriver.io/docs/api/visual-regression.html for more information."
`;

exports[`custom visual matcher > should allow asymmetric matchers 1`] = `
"Expected image to match with given asymmetric matcher but did not pass!
"
`;
